library/core/src/range/iter.rs

```diff
@@ -293,32 +294,74 @@ range_incl_exact_iter_impl! {
 /// By-value [`RangeFrom`] iterator.
 #[unstable(feature = "new_range_api", issue = "125687")]
 #[derive(Debug, Clone)]
-pub struct IterRangeFrom<A>(legacy::RangeFrom<A>);
+pub struct IterRangeFrom<A> {
```

Comment:
compiler impl looks reasonable to me. If you split out the libs changes to a separate PR I can approve this myself, otherwise needs libs approval too. I'm not sure where tests for libs stuff is supposed to go, I'm not certain that it's `tests/ui` but I honestly couldn't tell you